pcdsdevices/ipm.py

```diff
@@ -102,6 +105,48 @@ class IPMMotion(BaseInterface, Device):
     tab_whitelist = ['target', 'diode', 'insert', 'remove', 'inserted',
                      'removed']
 
+    def format_status_info(self, status_info):
+        """
+        Override status info handler to render the ipm.
+
+        Display ipm status info in the ipython terminal.
+
+        Parameters
+        ----------
+        status_info: dict
+            Nested dictionary. Each level has keys name, kind, and is_device.
+            If is_device is True, subdevice dictionaries may follow. Otherwise,
+            the only other key in the dictionary will be value.
+        Returns
+        -------
+        status: str
+            Formatted string with all relevant status information.
+        """
+        lines = []
+        name = ' '.join(self.prefix.split(':'))
+
+        x_motor_pos = status_info.get('diode', {}).get('x_motor', {}).get(
+                                      'position', 'N/A')
+        y_motor_pos = status_info.get('diode', {}).get('state', {}).get(
+                                      'motor', {}).get('position', 'N/A')
+        d_units = status_info.get('diode', {}).get('x_motor', {}).get(
+                                  'user_setpoint', {}).get('units', 'N/A')
+        target_pos = status_info.get('target', {}).get('motor', {}).get(
+                                     'position', 'N/A')
+        t_units = status_info.get('target', {}).get('motor', {}).get(
+                                  'user_setpoint', {}).get('units', 'N/A')
+        target_state_num = status_info.get('target', {}).get('state', {}).get(
+                                           'value', 'N/A')
+        target_state = status_info.get('target', {}).get('position', 'N/A')
+
+        name = f'{name}: Target {target_state_num} {target_state}'
```

Comment:
If I were you, I'd consider using the following pattern:
```suggestion
        return f"""\
{name}: Target {target_state_num} {target_state}
Target Position: {target_pos:.4f} [{t_units}]
Wave 8 Diode Position(x, y):
{x_motor_pos:.4f}, {y_motor_pos:.4f} [{d_units}]
"""
```

(wouldn't recommend "accepting" the suggestion, as I wrote this in the comment text box and not a proper editor)

Reply:
@klauer  thanks for the suggestions, is this what you meant 0efcae1 ?
This above prints:
```
In [3]: ipm
Out[3]:
XPP SB2 IPM: Target 5 OUT
Target Position: 0.0000 [mm]
Wave 8 Diode Position(x, y): 0.1063, -45.9985 [mm]
```
i will probably have to update the target position though, don't think i need a `.4f `there

Reply:
Yep, looks good. Purely a simplification/stylistic change.

Reply:
ok thanks, i'll change it for the others too